### ISSUE TITLE
performance improvement - split radiation into slow and fast routines

### ISSUE THREAD

**Issue by rgknox** [NGEET/fates#901]
From an algorithmic perspective, radiation scattering in the canopy is governed by slow (plant canopy composition and structure) and fast (incident radiation intensity and zenith angle) processes.

[subroutine PatchNormanRadiation](https://github.com/NGEET/fates/blob/sci.1.59.6_api.24.1.0/biogeophys/EDSurfaceAlbedoMod.F90#L181) is where we handle scattering.  This routine performs calculations that rely on both fast and slow.

Speaking generally, I think there is a way to split this routine so that we can preprocess as much of the slow timescale stuff as possible, once a day, and therefore reduce the number of computations that happen at sub-hourly calls (zenith stuff).  I'm not exactly sure how much things can be re-organized, but I think its worth a hard look.

**Comment by rosiealice** [comment]
I agree @rgknox. In general it could obviously do with an overhaul and there's lots that could be pulled out here. Plus there is still, I think, an issue with radiation conservation in non SP modes, and that might help improve things.
